### DISCUSSION TITLE
end2end example: Brevitas to deployment CNN?

### DISCUSSION THREAD

**Discussion by julienmas** [Xilinx/finn#669] [General]
Hi,

I was wondering if it exists an example of a model using convolutional layers coded with Brevitas somwhere, that can be simply transformed with FINN and deployed on a PYNQ board. All the examples I was able to find were already .onnx files. Lately I tried to used the brevitas LowPrecisionLeNet model described in the brevitas github repo's readme.md, but I wasn't able to build a PYNQ driver using DataflowBuildConfig() and build_dataflow_cfg() functions.

Julien

**Comment by rpitonak** [comment]
Hello,

Have you already seen [this notebook (cnv_end2end_example.ipynb)](https://github.com/Xilinx/finn/blob/main/notebooks/end2end_example/bnn-pynq/cnv_end2end_example.ipynb)?

It uses a model from [Brevitas example (CNV_1W1A)](https://github.com/Xilinx/brevitas/tree/master/src/brevitas_examples/bnn_pynq) with all the code available.

Hope it helps.

Rado

**Reply by julienmas** [reply]
Thanks, I wasn't able to find this Brevitas code ^^'
I'm still open to other examples though, this example uses `ModuleList()` to build the model, I was looking more for an example using the "classic" way to build a model (not using `Sequential` either). I know it sounds like details but it's always useful to have multiple examples to get to know FINN, especially for beginners :)
